src/XMakeBuildEngine/Definition/ProjectCollection.cs

```diff
@@ -384,7 +384,7 @@ public static Version Version
                     // Use .CodeBase instead of .Location, because .Location doesn't
                     // work when Microsoft.Build.dll has been shadow-copied, for example
                     // in scenarios where NUnit is loading Microsoft.Build.
-                    s_engineVersion = new Version(FileVersionInfo.GetVersionInfo(FileUtilities.ExecutingAssemblyPath).ProductVersion);
+                    s_engineVersion = new Version(FileVersionInfo.GetVersionInfo(FileUtilities.ExecutingAssemblyPath).FileVersion);
```

Comment:
AssemblyVersion is always a M.m.b[.r] format.
AssemblyFileVersion is also an M.m.b.r format
AssemblyInformationalVersion has the M.m.b+g1a4daf format.

I don't know what this GetVersionInfo API returns for ProductVersion or FileVersion though.